onnxruntime/core/framework/murmurhash3.cc

```diff
@@ -59,7 +59,7 @@ inline uint64_t rotl64(uint64_t x, int8_t r) {
 //
 // Changes to support big-endian from https://github.com/explosion/murmurhash/pull/27/
 // were manually applied to original murmurhash3 source code.
-ORT_FORCEINLINE uint32_t getblock32(const uint32_t* p, int i) {
+ORT_FORCEINLINE uint32_t getblock32(const uint32_t* p, long i) {
```

Comment:
Suggest to use ptrdiff_t instead of long.

Reply:
Should it be size_t instead? It should never be negative and the values passed in are mostly size_t already. Or are negative values useful? (since ptrdiff_t is size_t sized, but signed)

Reply:
The reason they use signed is because there is a loop inside.
 `for (auto i = -nblocks; i; i++) {`
`nblocks` has same type with len.

Reply:
updated to ptrdiff_t